src/SpendPermissionManager.sol

```diff
@@ -352,45 +356,45 @@ contract SpendPermissionManager is EIP712 {
         return allApproved;
     }
 
-    /// @notice Approves a permission while revoking another if its last update has not changed.
+    /// @notice Approve a spend permission while revoking another if its last updated period matches an expected value.
     ///
-    /// @dev Enforces that the last updated period of the permission being revoked matches the last valid updated period
-    ///      submitted as an argument. This is to prevent frontrunning `approveWithRevoke` with additional last-minute
-    ///      spends.
-    /// @dev The `account` of the permissions must match, but the remaining fields can differ.
-    /// @dev Can only be called by the `account` of a permission.
+    /// @dev Comparing the revoked spend permission's last updated period mitigates frontrunning with further spend.
+    /// @dev The accounts of the spend permissions must match, but all other fields can differ.
+    /// @dev Reverts if not called by the account of the spend permissions.
     ///
     /// @param permissionToApprove Details of the spend permission to approve.
     /// @param permissionToRevoke Details of the spend permission to revoke.
-    /// @param lastValidUpdatedPeriod Last valid updated period for the spend permission being revoked.
+    /// @param expectedLastUpdatedPeriod Expected last updated period for the spend permission being revoked.
```

Comment:
not official, but following our existing convention for "expected" comparison in errors